Add triggers documentation (#3448)
SumoLogic/sumologic-documentation: docs/cloud-soar/cloud-soar-integration-framework.md

   exit(0)

check_not_null_field: 'opt_92' #specify that the custom field with internal id “opt_92” should be populated in
order to display this button in the GUI.
hook:
 - incidentCustomAction #hook that generates a custom button in the incident tombstone when field opt_92 is not
null
```

For other example YAML files, see:
* [Trigger taskCustomAction definition file (Incident Tools)](/docs/cloud-soar/cloud-soar-integration-framework/#trigger-taskcustomaction-definition-file-incident-tools)
* [Trigger incidentCustomAction definition file (Incident Tools)](/docs/cloud-soar/cloud-soar-integration-framework/#trigger-incidentcustomaction-definition-file-incident-tools)

#### check_not_null_field

Specifying the YAML field `check_not_null_field : <field ID>` inside a trigger, you can ensure that the trigger is only executed (or displayed in case of custom actions buttons) when a specific incident field is populated.

For example, if we want to display our custom action trigger button only when our custom field with field ID `opt_15` is populated, inside the YAML file you can set:

```json
check_not_null_field: 'opt_15'
hook:
 - incidentCustomAction
```

#### Define a webhook trigger

> Triggers with the hook `webhook` specified can interact with the payload that is posted on your 
> webhook endpoint:

Author: ```suggestion
Triggers with the hook `webhook` specified can interact with the payload that is posted on your webhook endpoint:
```

   payload = {
      "additional_info": additional_info,
   }
   incident = session.put(end_point, headers=headers, data=payload, proxies=None, timeout=(5, 60))
   try:
   incident.raise_for_status()
   except Exception as e:
   sys.stderr.write(str(e))
   exit(0)

check_not_null_field: 'opt_92' #specify that the custom field with internal id “opt_92” should be populated in
order to display this button in the GUI.
hook:
 - incidentCustomAction #hook that generates a custom button in the incident tombstone when field opt_92 is not
null
```

For other example YAML files, see:
* [Trigger taskCustomAction definition file (Incident Tools)](/docs/cloud-soar/cloud-soar-integration-framework/#trigger-taskcustomaction-definition-file-incident-tools)
* [Trigger incidentCustomAction definition file (Incident Tools)](/docs/cloud-soar/cloud-soar-integration-framework/#trigger-incidentcustomaction-definition-file-incident-tools)

#### check_not_null_field

Specifying the YAML field `check_not_null_field : <field ID>` inside a trigger, you can ensure that the trigger is only executed (or displayed in case of custom actions buttons) when a specific incident field is populated.

> For example, if we want to display our custom action trigger button only when our custom field with
> field ID `opt_15` is populated, inside the YAML file you can set:

Author: ```suggestion
For example, if we want to display our custom action trigger button only when our custom field with field ID `opt_15` is populated, inside the YAML file you can set:
```

     * h => HOURS
     * m => MINUTES

### Trigger action definitions

Triggers allow you to extend Cloud SOAR functionality with custom logic using OIF actions. Triggers are invoked based on specific events that can manually be invoked by Cloud SOAR users, or invoked automatically by interacting with the appropriate endpoint of the API. For example, you can activate a trigger when a button is pressed, a field value is updated in an incident, or when new objects are created. 

#### Define a trigger action

To create a new trigger action, you must define a trigger action YAML file with the value `type` set to `Trigger`:

<img src={useBaseUrl('img/cloud-soar/trigger-code-example.png')} alt="Trigger action" width="800"/>

> Depending on the logic that you want to implement in your triggers, specify a list of one or more [hooks](#trigger-hooks) in the trigger YAML file. Each hook represents a manual event or API endpoint that can invoke the trigger. For example, by specifying the hook `updateIncident` inside a trigger, the trigger will fire whenever the field of any incident is updated either manually from the GUI or via the API.

Reviewer: ```suggestion
Depending on the logic that you want to implement in your triggers, specify a list of one or more [hooks](#trigger-hooks) in the trigger YAML file. Each hook represents a manual event or API endpoint that can invoke the trigger. For example, by specifying the hook `updateIncident` inside a trigger, the trigger will fire whenever the field of any incident is updated either manually from the UI or via the API.
```

> Triggers function as individual actions, executed in the backend, without the capability to review

Reviewer: Can you join this with line 530?

> Triggers function as individual actions, executed in the backend, without the capability to review
> the execution output in the GUI except for triggers on entities (observables). If a trigger fails, error logs printed on the `stderr` output of the trigger are exported in the audit trail (system log verbosity must be set to `ALL` to review trigger audit logs). Triggers cannot receive manual input, except for [triggers with the `incidentCustomActions` and `taskCustomActions`hooks](#trigger-incidentcustomaction-and-taskcustomaction) that accept a text input.

Author: ```suggestion
Triggers function as individual actions, executed in the backend, without the capability to review the execution output in the GUI except for triggers on entities (observables). If a trigger fails, error logs printed on the `stderr` output of the trigger are exported in the audit trail (system log verbosity must be set to `ALL` to review trigger audit logs). Triggers cannot receive manual input, except for [triggers with the `incidentCustomActions` and `taskCustomActions`hooks](#trigger-incidentcustomaction-and-taskcustomaction) that accept a text input.
```

   payload = {
      "additional_info": additional_info,
   }
   incident = session.put(end_point, headers=headers, data=payload, proxies=None, timeout=(5, 60))
   try:
   incident.raise_for_status()
   except Exception as e:
   sys.stderr.write(str(e))
   exit(0)

check_not_null_field: 'opt_92' #specify that the custom field with internal id “opt_92” should be populated in
order to display this button in the GUI.
hook:
 - incidentCustomAction #hook that generates a custom button in the incident tombstone when field opt_92 is not
null
```

For other example YAML files, see:
* [Trigger taskCustomAction definition file (Incident Tools)](/docs/cloud-soar/cloud-soar-integration-framework/#trigger-taskcustomaction-definition-file-incident-tools)
* [Trigger incidentCustomAction definition file (Incident Tools)](/docs/cloud-soar/cloud-soar-integration-framework/#trigger-incidentcustomaction-definition-file-incident-tools)

#### check_not_null_field

Specifying the YAML field `check_not_null_field : <field ID>` inside a trigger, you can ensure that the trigger is only executed (or displayed in case of custom actions buttons) when a specific incident field is populated.

> For example, if we want to display our custom action trigger button only when our custom field w ith

Reviewer: ```suggestion
For example, if we want to display our custom action trigger button only when our custom field with
```

> field ID `opt_15` is populated, inside the YAML file you can set:

Reviewer: Can you join this with line 719?

```json
check_not_null_field: 'opt_15'
hook:
 - incidentCustomAction
```

#### Define a webhook trigger

> Triggers with the hook `webhook` specified can interact with the payload that is posted on your 

Reviewer: Pls join with next line
